README.md

```diff
@@ -37,6 +37,32 @@ AMD SEV-SNP API formatted report and certificate table, or just `GetReport`,
 `GetReportAtVmpl`, `GetRawReport`, or `GetRawReportAtVmpl` to avoid fetching the
 certificate table.
 
+### `func GetDerivedKeyAcknowledgingItsLimitations(d Device, request *SnpDerivedKeyReq) ([]byte, error)`
+
+This function uses the `/dev/sev-guest` command for requesting a key derived
+from data that is measured at VM launch time, with the additional ability to
+continue to generate the same key as at earlier TCB and GuestSVN values.
+
+This function's name is selected to discourage its use in a Cloud setting. When
+you do not have full control over the machine that derives the key (e.g., a
+Cloud Service Provider (CSP) runs multiple customers' jobs on the same machine),
+you should not set `UseVCEK` to true. This is because all of the data measured
+at launch on a CSP will likely be shared across multiple customers by means of a
+CSP-provided firmware, and therefore other customers that happen to run on the
+same machine can get access to the same derived keys. Firmware binaries and
+other launch information are not secrets either, so the security entirely hinges
+on the root key.
+
+The migration agent (MA)-based VMRK behavior is also subject to the limitations
```

Comment:
The goal of MSG_KEY_REQ is to create a core primitive to build sealing keys. Sealing keys allow only the sealing software to unseal at a later date, even if the instance was torn down and rebuilt. When you base your keys on VMRK, you lose that property since VMRK is registered by the MA on first launch, usually randomly. If you want the same key, you need to solve the seal-to-workload problem to persist a key for the MA to get and register as a same VMRK if you want that key to be secure.